### PR TITLE
Orca integration

### PR DESCRIPTION
**Pull request by edenorca**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Integration to Orca Security (Cloud)

## Screenshots
image

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [X] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/Orca/pack_metadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "url": "https://www.paloaltonetworks.com/cortex",
+    "email": "support@orca.security",
+    "created": "2020-12-20T17:38:25Z",
+    "categories": [
+        "Vulnerability Management"
+    ],
+    "tags": [],
+    "useCases": [],
+    "keywords": []
+}
```
Comment (ChanochShayner) [inline Packs/Orca/pack_metadata.json:16]: In order to be notified of any changes to your pack, please add the key - `githubUser` to this metadata file with your Github User.

File: Packs/Orca/pack_metadata.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "Orca",
+    "description": "Orca Security cloud integration",
+    "support": "partner",
+    "currentVersion": "1.0.0",
+    "author": "Cortex XSOAR",
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (ChanochShayner) [inline Packs/Orca/pack_metadata.json:7]: Note that this should be your details.

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "cliName": "cloudaccount",
+    "closeForm": false,
+    "content": true,
+    "editForm": true,
+    "group": 0,
+    "hidden": false,
+    "id": "incident_cloudaccount",
+    "isReadOnly": false,
+    "locked": false,
+    "name": "Cloud Account",
```
Comment (ChanochShayner) [inline Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:16]: I think it's worth changing the name to -
```suggestion
    "name": "Orca Cloud Account",
```

**Review by ChanochShayner — COMMENTED** [review]
Thank you for your contribution, it looks great!

For the README file, you can use `demisto-sdk generate-docs`
https://xsoar.pan.dev/docs/integrations/integration-docs

**Comment by ChanochShayner** [comment]
All the entities you have should have the same `fromversion`,
At the moment they are all 5.0.0 and only the classifier and mapper are 6.0.0 because there is support for this type of classifier and mapper only version 6.0.0.
It has two possible solutions:
1 - Move all entities to version 6.0.0
2 - If you want this pack to be supported from version 5.0.0 then you need to add another classifier (which also includes a mapper) - https://xsoar.pan.dev/docs/contributing/checklist#classifiers-and-mappers

File: Packs/Orca/Integrations/Orca/Orca.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+- display: Incident type
+  name: incidentType
+  required: false
+  type: 13
+- display: Fetch incidents
+  name: isFetch
+  required: false
+  type: 8
+- defaultvalue: "200"
+  display: ""
```
[the remark this answers is not in the document]
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/Orca.yml:24]: done

File: Packs/Orca/Integrations/Orca/Orca.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    outputs:
+    - contextPath: Orca.Manager.Asset
+      description: Asset description
+      type: String
+  - arguments:
+    - description: Orca Asset Unique Identifier
+      name: asset_unique_id
+      required: true
+    description: Get CVEs of an asset
+    name: get-cves
```
[the remark this answers is not in the document]
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/Orca.yml:66]: I removed it from the yaml
done

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json (hunk 3 above)
Comment (edenorca) [reply Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:16]: done

File: Packs/Orca/pack_metadata.json (hunk 2 above)
Comment (edenorca) [reply Packs/Orca/pack_metadata.json:7]: done

File: Packs/Orca/pack_metadata.json (hunk 1 above)
Comment (edenorca) [reply Packs/Orca/pack_metadata.json:16]: done

File: Packs/Orca/Integrations/Orca/Orca.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        client = BaseClient(
+            base_url=ORCA_API_DNS_NAME,
+            verify=True,
+            headers={
+                'Authorization': f'Bearer {api_key}'
+            },
+            proxy=True)
+
+        orca_client = OrcaClient(client=client)
+        if command == "get-alerts":
```
[the remark this answers is not in the document]
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/Orca.py:123]: done

**Comment by edenorca** [comment]
> All the entities you have should have the same `fromversion`,
> At the moment they are all 5.0.0 and only the classifier and mapper are 6.0.0 because there is support for this type of classifier and mapper only version 6.0.0.
> It has two possible solutions:
> 1 - Move all entities to version 6.0.0
> 2 - If you want this pack to be supported from version 5.0.0 then you need to add another classifier (which also includes a mapper) - https://xsoar.pan.dev/docs/contributing/checklist#classifiers-and-mappers

done

File: Packs/Orca/Integrations/Orca/Orca.yml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  type: 13
+- display: Fetch incidents
+  name: isFetch
+  required: false
+  type: 8
+- defaultvalue: "200"
+  display: ""
+  name: max_fetch
+  required: false
+  type: 0
```
[the remark this answers is not in the document]
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/Orca.yml:26]: thank you, we discussed this.

File: Packs/Orca/Integrations/Orca/Orca.yml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  type: 8
+description: Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS,
+  Azure, and GCP.
+display: Orca
+name: Orca
+script:
+  commands:
+  - arguments:
+    - description: Type of alert to get
+      name: alert_type
```
[the remark this answers is not in the document]
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/Orca.yml]: the type of alerts are dynamic and change between organizations.
I don't have a closed list of the alert types

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "6.0.0"
+}
```
Comment (ChanochShayner) [inline Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:29]: In this file you need to change the name of the file - 
to incidentfield-Orca_Cloud_Account.json

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        "Orca Alert"
+    ],
+    "caseInsensitive": true,
+    "cliName": "cloudaccount",
+    "closeForm": false,
+    "content": true,
+    "editForm": true,
+    "group": 0,
+    "hidden": false,
+    "id": "incident_cloudaccount",
```
Comment (ChanochShayner) [inline Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:13]:
```suggestion
    "id": "incident_orcacloudaccount",
```

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json [hunk 11]
```diff
@@ -0,0 +1,29 @@
+{
+    "associatedToAll": false,
+    "associatedTypes": [
+        "Orca Alert"
+    ],
+    "caseInsensitive": true,
+    "cliName": "cloudaccount",
```
Comment (ChanochShayner) [inline Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:7]:
```suggestion
    "cliName": "orcacloudaccount",
```

File: Packs/Orca/Integrations/Orca/README.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
+Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS, Azure, and GCP.
+This integration was integrated and tested with version xx of Orca
```
Comment (ChanochShayner) [inline Packs/Orca/Integrations/Orca/README.md:2]:
```suggestion
This integration was integrated and tested with version ? of Orca
```
There is specific version?
if not, remove it.

File: Packs/Orca/Integrations/Orca/README.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+| --- | --- | --- |
+| Orca.Manager.Asset | String | Asset description | 
+
+
+#### Command Example
+``` ```
+
+#### Human Readable Output
+
+
```
Comment (ChanochShayner) [inline Packs/Orca/Integrations/Orca/README.md:80]: Please remove it.

File: Packs/Orca/Integrations/Orca/README.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+| Orca.Manager.Alerts | String | All alerts | 
+
+
+#### Command Example
+``` ```
+
+#### Human Readable Output
+
+
+
```
Comment (ChanochShayner) [inline Packs/Orca/Integrations/Orca/README.md:52]: please remove it

**Review by ChanochShayner — COMMENTED** [review]
Looks great,
Are you using test_data files?
If so, it's worth making sure there are no confidential things.
If not, you can remove them.

File: Packs/Orca/Integrations/Orca/README.md (hunk 13 above)
Comment (ChanochShayner) [reply Packs/Orca/Integrations/Orca/README.md:80]:
```suggestion
#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| Orca.Manager.Asset | String | Asset description | 

```

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json (hunk 9 above)
Comment (edenorca) [reply Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json:29]: done

**Comment by edenorca** [comment]
> Looks great,
> Are you using test_data files?
> If so, it's worth making sure there are no confidential things.
> If not, you can remove them.

 done

File: Packs/Orca/Integrations/Orca/README.md (hunk 14 above)
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/README.md:52]: done

File: Packs/Orca/Integrations/Orca/README.md (hunk 13 above)
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/README.md:80]: done

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json [hunk 15]
```diff
@@ -0,0 +1,78 @@
+{
+    "description": "Maps Orca fields for use in Integration Playbooks.",
+    "feed": false,
+    "id": "Orca Mapper",
+    "mapping": {
+        "Orca Alert": {
+            "dontMapEventToLabels": false,
+            "internalMapping": {
+                "Cloud Acccount": {
```
Comment (ChanochShayner) [inline Packs/Orca/Classifiers/classifier-Orca_Mapper.json:9]:
```suggestion
                "Orca Cloud Acccount": {
```

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+                    "complex": null,
+                    "simple": "cloud_provider_id"
+                },
+                "Orca Alert ID": {
+                    "complex": null,
+                    "simple": "state.alert_id"
+                },
+                "Orca Alert Id": {
+                    "complex": null,
+                    "simple": "state.alert_id"
```
Comment (ChanochShayner) [inline Packs/Orca/Classifiers/classifier-Orca_Mapper.json:19]: Remove,  duplicate..

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+                    "complex": null,
+                    "simple": "asset_unique_id"
+                },
+                "Orca Reason": {
+                    "complex": null,
+                    "simple": "type_string"
+                },
+                "Orca_Asset_Unique_ID": {
+                    "complex": null,
+                    "simple": "asset_unique_id"
```
Comment (ChanochShayner) [inline Packs/Orca/Classifiers/classifier-Orca_Mapper.json:31]: remove, duplicate ..

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+                },
+                "cloudaccount": {
+                    "complex": null,
+                    "simple": "account_name"
```
Comment (ChanochShayner) [inline Packs/Orca/Classifiers/classifier-Orca_Mapper.json:35]: remove, duplicate ..?

**Review by ChanochShayner — COMMENTED** [review]
There are some duplicates fields in the mapper:

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json (hunk 10 above)
Comment (edenorca) [reply Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json]: done

File: Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json (hunk 11 above)
Comment (edenorca) [reply Packs/Orca/IncidentFields/incidentfield-Cloud_Account.json]: done

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json (hunk 17 above)
Comment (edenorca) [reply Packs/Orca/Classifiers/classifier-Orca_Mapper.json:31]: done

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json (hunk 16 above)
Comment (edenorca) [reply Packs/Orca/Classifiers/classifier-Orca_Mapper.json:19]: done

File: Packs/Orca/Classifiers/classifier-Orca_Mapper.json (hunk 18 above)
Comment (edenorca) [reply Packs/Orca/Classifiers/classifier-Orca_Mapper.json:35]: done

File: Packs/Orca/Integrations/Orca/README.md (hunk 12 above)
Comment (edenorca) [reply Packs/Orca/Integrations/Orca/README.md]: done

**Comment by ChanochShayner** [comment]
There was a demo with Orca this morning,
it was great!
In conclusion,
We will wait for Orca to complete the process of building the incident stream.
We might want to add some more configuration information.
Because of the unfamiliar form of fetch incident, it is worth adding an explanation of the process.
In case that alert does not exist - we would like to return such a message and not return an error.
Check the option to add a filter on the Orca side to sort the incoming alerts.

**Review by ChanochShayner — APPROVED** [review]
There are some duplicates fields in the mapper:

**Pull request merged**